### PR TITLE
[to#issue-817]Add support for Clickzetta engine

### PR DESCRIPTION
**Pull request by idling11**
Add clickzetta driver for data-diff

### CODE REVIEW

**Comment by idling11** [comment]
I am confused that I have clickzetta-connector == 0.8.59 in pypi,  why the check is failed when download clickzetta-connector. The Download source is not pypi?  
image

**Comment by sungchun12** [comment]
Setting a reminder to review this PR later this week!

**Comment by dlawin** [comment]
> I am confused that I have clickzetta-connector == 0.8.59 in pypi, why the check is failed when download clickzetta-connector. The Download source is not pypi?

Can you run `poetry lock --no-update` ? That will add the dependency to the lockfile

**Comment by idling11** [comment]
@sungchun12 I used our test account to test the corresponding ut, and they were all passed. It is currently in the external whitelisting stage and is expected to be ready on January 10. After it is available, we will run it through CI. Thank you so much

**Comment by sungchun12** [comment]
@idling11 Can you open source the code for the `clickzetta-connector` and its license? We'll need it for security review and submit issues if it's not working well with `data-diff` over time.

File: tests/test_clickzetta.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        a = TableSegment(self.connection, self.table_src.path, ("id",), "comment")
+        b = TableSegment(self.connection, self.table_dst.path, ("id",), "comment")
+
+        differ = HashDiffer()
+        diff = list(differ.diff_tables(a, b))
+        id = diff[0][1][0]
+        self.assertEqual(diff, [("-", (id, "This one is different"))])
+
+        # Compare with MySql
+        mysql_conn = get_conn(db.MySQL)
```
Comment (sungchun12) [inline tests/test_clickzetta.py:45]: Can you add tests for other common databases clickzetta is using data-diff with? Ex: PostgreSQL

File: tests/test_clickzetta.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            self.table_src.insert_row(200, "This one is different", columns=["id", "comment"]),
+        ]
+
+        for query in queries:
+            self.connection.query(query)
+
+        a = TableSegment(self.connection, self.table_src.path, ("id",), "comment")
+        b = TableSegment(self.connection, self.table_dst.path, ("id",), "comment")
+
+        differ = HashDiffer()
```
Comment (sungchun12) [inline tests/test_clickzetta.py:39]: Add `Clickzetta` to the joindiff algorithm as I imagine people want to run in-db diffs too:

- https://github.com/datafold/data-diff/blob/a2c64acba5ebcc8454db647abb3d0e83031c095b/data_diff/joindiff_tables.py#L12
- https://github.com/datafold/data-diff/blob/a2c64acba5ebcc8454db647abb3d0e83031c095b/data_diff/joindiff_tables.py#L160

**Comment by sungchun12** [comment]
I'll wait until Jan 10 to provide fuller feedback.

**Comment by idling11** [comment]
> @idling11 Can you open source the code for the `clickzetta-connector` and its license? We'll need it for security review and submit issues if it's not working well with `data-diff` over time.

OK, I have submitted it to the internal approval process of open source. I will give you the link after the approval is passed.

File: tests/test_clickzetta.py (hunk 1 above)
Comment (idling11) [reply tests/test_clickzetta.py:51]: Added PG check

File: tests/test_clickzetta.py (hunk 2 above)
Comment (idling11) [reply tests/test_clickzetta.py:45]: Added joindiff and UT for clickzetta

**Pull request closed without merging**